### PR TITLE
Expand RFC Pattern

### PR DESCRIPTION
**Pull request by tsadler1988**
Addresses #215:
- Story added
- Known Instances expanded 
- Expand on Solutions and Resulting Context
- Add Aliases
- Some (not all) feedback from #204 addressed

### CODE REVIEW

**Comment by tsadler1988** [comment]
Not sure why the checks are failing - any advice would be appreciated 🙂

**Comment by spier** [comment]
@tsadler1988 Thank you for doing this work. Looks fantastic on a first glance.

I will make sure to give this a proper review, so that we can level up the maturity of the pattern and (most importantly for me right now) get it released in the book 📘 

> Not sure why the checks are failing - any advice would be appreciated 🙂

Nothing to worry about regarding the checks, I will fix them!

File: patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md
Comment (spier) [inline patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md]:
```suggestion
- Writing up design proposals (architecture, protocols, etc) upfront has an element of waterfall-like design that does not fit the iterative development approach that many development teams favor. Remember: “Working software over comprehensive documentation” ([Agile Manifesto](https://agilemanifesto.org/)). The RFC process should be as lightweight as possible.
```

Adding a link. Many people will already know the manifesto but hey, doesn't hurt :)

Also shortening the text a bit.

File: patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md
Comment (spier) [inline patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md]:
```suggestion
## Aliases
```

Reducing level of heading.

File: patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md
Comment (spier) [inline patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md]:
```suggestion
- **BBC iPlayer & Sounds** - As presented at the ISC Fall Summit 2020 [Using Internal RFCs to Enhance Collaboration][bbc].
- **Europace** - As described in Open Organization: [Setting cross-team standards and best practices in the open][europace].
- **Uber** - According to this blog post by Gergely Orosz: [Scaling Engineering Teams via Writing Things Down and Sharing - aka RFCs][uber].
- **Google Design Docs** - As described in this blog post by Malte Ubl [Design Docs at Google][google]
```

I vote for calling out the BBC first, as you are the one putting in the work to contribute this pattern.

File: patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md
Comment (spier) [inline patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md]:
```suggestion
- Architecture Decision Record (ADRs) - Not necessarily a direct alias, as they can sometimes be used very differently e.g. RFCs for seeking input and building consensus, ADRs for recording decisions and implementation detail
```

Simplifying the list.

**Review by spier — CHANGES_REQUESTED** [review]
Great iteration! I left a couple of comments inline.

I would be happy to merge this when the comments are addressed. I will leave some additional ideas in a separate comment.

**Comment by spier** [comment]
@tsadler1988 again, you produced a much better version than what we had before already.

Here some some ideas for things that would make the pattern even more attractive in the book:

- **a shorter title**: would make it easier to reference the pattern (e.g. from other patterns), and shorter titles also look better in the table of contents in the book :) Current title is "Transparent Cross-Team Decision Making using RFCs". Some ideas for shorter titles "Decision Making using RFCs", "Internal RFCs"

- **a sketch**: any type of visual really makes the pattern prettier in the book. Could even be a screenshot of a mockup RFC template? or just a screenshot of one of the RFCs from RUST or elsewhere?

- **sharing a template**: The [Base Documentation](https://innersourcecommons.gitbook.io/innersource-patterns/patterns/base-documentation) pattern includes two templates, than can help people to start adopting the pattern faster. Do you think the at a generalized version of the RFC template from the RFC could be shared? Otherwise I can maybe try to put something together based on the internal-RFCs that we use at Meltwater.

**Comment by spier** [comment]
@tsadler1988 if you merge in the latest changes from master, then the checks should be fixed.

**Comment by spier** [comment]
Hmm. Looks like now the checks don't run on your branch at all anymore. That's also a way to fix them 🤣 

Anyhow, let's focus this PR on the content of your pattern, and I will take care of the checks elsewhere.

**Comment by spier** [comment]
@tsadler1988 if you like you can merge `master` into your branch again, and how you should see the LinkChecker execute correctly. :) Sorry for the troubles.

File: patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md
Comment (spier) [reply patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md]: This should be non-controversial, so I am committing this suggestion.

**Comment by spier** [comment]
@tsadler1988 hope I didn't scare you aware with all my feedback. :)

If you like, then we can shorten this iteration a bit and only do these two things for now:
- review my inline comments in this PR
- finding a shorter title if at all possible - see [comment](https://github.com/InnerSourceCommons/InnerSourcePatterns/pull/245#issuecomment-744686082) 

Also I hope you have a couple of relaxing days during this holiday season.

**Review by lenucksi — APPROVED** [review]
LGTM after getting the discussions resolved. Also, I'm good with the length of the title. Its concise and descriptive.

Also, love having the rich list of examples! :smile:

File: patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md
Comment (spier) [inline patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md:114]:
```suggestion
```

Removing one newline to make the linter happy.

**Comment by spier** [comment]
@tsadler1988 if you don't have any time right now, just give me a quick shout. I will then resolve my feedback in the least invasive way, and get your PR merged, because I don't want to hide this great pattern improvement in this PR.

Also note to self:
Remove the faulty link to `patterns/1-initial/transparent-cross-team-decision-making-using-rfcs.md` from README.md which I introduced myself by making a mistake when merging in master. (this is why the "Link Check" is failing)

File: patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md
Comment (tsadler1988) [reply patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md]: Good suggestion :+1:

File: patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md
Comment (tsadler1988) [reply patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md]: Sure, happy to take this change :)

**Comment by tsadler1988** [comment]
Apologies for disappearing these last 4 weeks, I should've dropped you a message, but I'm back now. 

Thanks for fixing the checks and the reviews, I've taken the suggestions. Outstanding points as I understand it:
- **Title** - looks like this was discussed in the Patterns group and you're happy with it as is?
- **Template** - Good suggestion, I will add one in shortly.
- **Sketch** - Another good suggestion, but I would defer this to a future improvement so you can merge this iteration. I was thinking about an image of the template, but I think text is better as it can be copy + pasted. Perhaps a process flow diagram would be good?

**Comment by spier** [comment]
Thanks @tsadler1988. 

If I understand you correctly, I should wait for the Template to be added, and then merge this PR?

Alternatively I am also completely happy to merge this as is, and you can do any further changes at a later point via new PRs. 

Just let me know how you prefer to do this :)

File: patterns/2-structured/templates/rfc.md
Comment (spier) [inline patterns/2-structured/templates/rfc.md]:
```suggestion
```

File: patterns/2-structured/templates/rfc.md
Comment (spier) [inline patterns/2-structured/templates/rfc.md:70]:
```suggestion
## Unresolved questions

```

File: patterns/2-structured/templates/rfc.md
Comment (spier) [inline patterns/2-structured/templates/rfc.md:72]:
```suggestion

- What parts of the design do you expect to resolve through the RFC process before this gets merged?
```

**Comment by spier** [comment]
@tsadler1988 we have a markdownlinter now, as you noticed :)

I will let you review the checks/annotations I don't get in your way on this any further. Didn't notice that you were working on this at the same time as I was.

**Comment by tsadler1988** [comment]
> If I understand you correctly, I should wait for the Template to be added, and then merge this PR?

I think so! This is done now so I think it's ready to merge.

> I will let you review the checks/annotations I don't get in your way on this any further. Didn't notice that you were working on this at the same time as I was.

Haha yes, everything is passing now 🙂

**Comment by spier** [comment]
This is fantastic work @tsadler1988! I merged it, and we can do any other improvements that we want to do in new PRs.

The pattern is now live at https://innersourcecommons.gitbook.io/innersource-patterns/patterns/transparent-cross-team-decision-making-using-rfcs as well.

**Pull request merged**
